content/post/2023-01-24-code-switch-escape-hatch-test/index.Rmd

```diff
@@ -0,0 +1,116 @@
+---
+slug: code-switch-escape-hatch-test
+title: "A testing pattern: adding switches to your code" 
+authors: 
+- Maëlle Salmon 
+date: "2023-01-24" 
+tags: 
+- package development 
+output: hugodown::hugo_document
+---
+
+```{r setup, include=FALSE}
+
+knitr::opts_chunk$set(fig.path = "", comment = "")
+# knitr hook to make images output use Hugo options
+knitr::knit_hooks$set(
+  plot = function(x, options) {
+    hugoopts <- options$hugoopts
+    paste0(
+      "{{<figure src=",
+      '"', x, '" ',
+      if (!is.null(hugoopts)) {
+        glue::glue_collapse(
+          glue::glue('{names(hugoopts)}="{hugoopts}"'),
+          sep = " "
+        )
+      },
+      ">}}\n"
+    )
+  }
+)
+
+```
+
+Sometimes, testing [gets hard](https://r-pkgs.org/testing-advanced.html#when-testing-gets-hard).
+For instance, you'd like to test for the behavior of your function in the absence of an internet connection, or in an interactive session, without actually cutting off the internet, or from the safety of a definitely non interactive R session for tests.
+In this post we shall present a not too involved pattern to avoid very complicated infrastructure, as a complement to [mocking](/2019/10/29/mocking/) in your toolbelt.
```

Comment:
Would it be useful to add a small section about why / when you should prefer mocking to this pattern? The answer could be "mocking is always better but maybe not always worth the effort". But the question then becomes "at which point do you think it becomes useful / necessary to invest in getting mocking to work"?